PWG/EMCAL/EMCALtasks/AliEmcalAodTrackFilterTask.cxx

```diff
@@ -247,21 +235,32 @@ void AliEmcalAodTrackFilterTask::UserExec(Option_t *)
       return;
     }
   }
-
+*/
+AliParticleContainer* OutputCont= static_cast<AliParticleContainer*>(fParticleCollArray.Last());
   // add tracks to event if not yet there
   fTracksOut->Delete();
-  if (!(InputEvent()->FindListObject(fTracksOutName))) {
+  if(OutputCont->GetIsEmbedding()){
+        AliVEvent *event = AliEmcalContainerUtils::GetEvent(InputEvent(),kTRUE);
+        if(!(event->FindListObject(fTracksOutName))){
+        event->AddObject(fTracksOut);
+        OutputCont->SetArray(event);}
+        }
+  else if (!(InputEvent()->FindListObject(fTracksOutName))) {
     InputEvent()->AddObject(fTracksOut);
+        OutputCont->SetArray(InputEvent());
   }
-
+ AliAnalysisTaskEmcal::ExecOnce();
```

Comment:
This may or may not work, but it's probably a dangerous thing to do. `AliAnalysisTaskEmcal` does a lot of work behind the scenes for the user. I would strongly recommend that you change `UserExec` -> `Run` for this function to avoid being surprised by potentially broken`AliAnalysisTaskEmcal` functionality

Reply:
This line is certainly needed for the code to work. At the moment I am not sure what the differences between UserExec() and Run() are but will look into it. Do they generally serve the same purpose?

Reply:
I think I understand your comment a bit better now so I made replaced the UserExec() here with the combination of Run() and ExecOnce() similar to other modern tasks. Please have a look at commit e494991